### PR TITLE
HAL : add get last-week steps in environment

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
We can bring the data 7 days ago.

### CODE REVIEW

File: src/hal/environment.cpp [hunk 1]
```diff
@@ -184,6 +184,12 @@ uint32_t OswHal::Environment::getStepsOnDay(uint8_t dayOfWeek) {
         return this->getStepsToday();
     return this->_stepsCache[dayOfWeek];
 }
+uint32_t OswHal::Environment::getStepsLastWeek() {
+    uint32_t currDoM = 0; // Unused, but required by function signature
+    uint32_t currDoW = 0;
+    OswHal::getInstance()->getLocalDate(&currDoM, &currDoW);
+    return this->_stepsCache[currDoW];
```
Comment (simonmicro) [inline src/hal/environment.cpp:191]: But... The function signature implies the step amount of the whole week, but instead you return the current days values (i know, theoretically this is the value from last week). I think you should change the function name or expand the `getStepsOnDay()` with a special value like `8` to access this instead.

**Review by simonmicro — CHANGES_REQUESTED** [review]

File: src/hal/environment.cpp (hunk 1 above)
Comment (RuffaloLavoisier) [reply src/hal/environment.cpp:191]: > I think you should change the function name

Under what name?

Comment (RuffaloLavoisier) [reply src/hal/environment.cpp:191]: Now name `getStepsLastWeek()`, If I were to change function name, what would it be? Am I right to understand?

Comment (simonmicro) [reply src/hal/environment.cpp:191]: Maybe `getStepsBefore7Days()`? I think I would prefer enhancing the aforementioned existing function.

Any input @uvwxy ?

Comment (RuffaloLavoisier) [reply src/hal/environment.cpp:191]: @simonmicro 
> I think I would prefer enhancing the aforementioned existing function.

So Is it right to do it with a special numbers `8`? Or Is it better to embed default parameters`getStepsOnDay(uint8_t day, bool Before7Days=false)`?

Comment (RuffaloLavoisier) [reply src/hal/environment.cpp:191]: If we are approaching with a special number, we need to change that much logic.
If we're trying to access it, the `for-loop`(Mainly accessible methods) that you usually use will have more unnecessary loops.
In my humble opinion, wouldn't the default parameters be the most appropriate?

Comment (simonmicro) [reply src/hal/environment.cpp]: The default parameter is also a great idea, but you won't be able to return any other step-value except for exactly the current day of last week? So, basically you have to return 0 or throw an exception for any other case... Still, this is a pretty good idea (best so far to solve this) - especially as it preserves the current interface design and allows for later expansion!

**Review by simonmicro — APPROVED** [review]

**Pull request merged**